src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -133,20 +132,6 @@ export default function getAdaptationSwitchStrategy(
     return { type: "needs-reload", value: undefined };
   }
 
-  if (adaptation.type === "audio" &&
-      segmentBuffer.codec !== undefined &&
-      // We have been explicitly asked to reload
-      options.audioTrackSwitchingMode === "direct" &&
-      // We're playing the current Period
-      isTimeInRange({ start, end }, currentTime) &&
```

Comment:
We should still check that this is about the current period to avoid unnecessary flushing when e.g. changing the track for another one, no?

Though this can cause a problem if we are very close to the next Period - for which we just changed the audio track. Here we risk to not perform that flush and to obtain in consequence the previous track on some devices